cloudbuild.yaml

```diff
@@ -115,10 +115,21 @@ steps:
   args: ['make', '_GCB_POST_SUBMIT=${_GCB_POST_SUBMIT}', '_GCB_LATEST_VERSION=${_GCB_LATEST_VERSION}', 'SHORT_SHA=${SHORT_SHA}', 'BRANCH_NAME=${BRANCH_NAME}', 'push-images', '-j8']
   waitFor: ['Build: Assets']
 
+- id: 'Test: Terraform Configuration'
+  name: 'gcr.io/$PROJECT_ID/open-match-build'
+  args: ['make', 'terraform-test']
+  waitFor: ['Build: Install Toolchain']
+
+- id: 'Test: Create Cluster'
+  name: 'gcr.io/$PROJECT_ID/open-match-build'
+  args: ['make', 'SHORT_SHA=${SHORT_SHA}', 'delete-gke-cluster', 'create-gke-cluster', 'push-helm', 'update-chart-deps']
+  waitFor: ['Build: Install Kubernetes Tools']
+
+# TODO: `make update-chart-deps` requires `helm init`. This requirement will go away after introducing helm3.
 - id: 'Build: Deployment Configs'
   name: 'gcr.io/$PROJECT_ID/open-match-build'
   args: ['make', 'VERSION_SUFFIX=$SHORT_SHA', 'clean-install-yaml', 'install/yaml/']
-  waitFor: ['Build: Install Toolchain']
+  waitFor: ['Test: Create Cluster']
```

Comment:
It turns out that helm will use the `tar.gz` packages under the /charts directory for `helm install/upgrade` commands, so any updates to the templates won't be reflected without repacking the packages using `make update-charts-dep`. The dependency at here is changing because `make update-charts-dep` requires a tiller pod, which is created by `make push-helm` in the `Test: Create Cluster` step. We can change this line back once helm3 is released.